FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeProcessingJobSpliter.cs

```diff
@@ -0,0 +1,291 @@
+        }
+
+        // get the lastUpdated timestamp of next resource for next processing job
+        private async Task<DateTimeOffset?> GetNextTimestamp(string resourceType, DateTimeOffset? start, DateTimeOffset end, bool isDescending, CancellationToken cancellationToken)
+        {
+            List<KeyValuePair<string, string>> parameters = new List<KeyValuePair<string, string>>
+            {
+                new KeyValuePair<string, string>(FhirApiConstants.LastUpdatedKey, $"lt{end.ToInstantString()}"),
+                new KeyValuePair<string, string>(FhirApiConstants.PageCountKey, FhirApiPageCount.Single.ToString("d")),
+                new KeyValuePair<string, string>(FhirApiConstants.TypeKey, resourceType),
+            };
+
+            if (isDescending)
+            {
+                parameters.Add(new KeyValuePair<string, string>(FhirApiConstants.SortKey, "-_lastUpdated"));
+            }
+            else
+            {
+                parameters.Add(new KeyValuePair<string, string>(FhirApiConstants.SortKey, "_lastUpdated"));
+            }
+
+            if (start != null)
+            {
+                parameters.Add(new KeyValuePair<string, string>(FhirApiConstants.LastUpdatedKey, $"ge{((DateTimeOffset)start).ToInstantString()}"));
+            }
+
+            var searchOptions = new BaseSearchOptions(null, parameters);
+
+            string fhirBundleResult = await _dataClient.SearchAsync(searchOptions, cancellationToken);
+
+            // Parse bundle result.
+            JObject fhirBundleObject;
+            try
+            {
+                fhirBundleObject = JObject.Parse(fhirBundleResult);
+            }
+            catch (JsonReaderException exception)
+            {
+                string reason = string.Format(
+                        "Failed to parse fhir search result for '{0}' with search parameters '{1}'.",
+                        searchOptions.ResourceType,
+                        string.Join(", ", searchOptions.QueryParameters.Select(parameter => $"{parameter.Key}: {parameter.Value}")));
+
+                _diagnosticLogger.LogError(reason);
+                _logger.LogInformation(exception, reason);
+                throw new FhirDataParseException(reason, exception);
+            }
+
+            List<JObject> fhirResources = FhirBundleParser.ExtractResourcesFromBundle(fhirBundleObject).ToList();
+
+            if (fhirResources.Any() && fhirResources.First().GetLastUpdated() != null)
+            {
+                return fhirResources.First().GetLastUpdated();
+            }
+
+            return null;
+        }
+
+        private async Task<DateTimeOffset?> GetNextSplitTimestamp(string resourceType, DateTimeOffset? start, Dictionary<DateTimeOffset, int> anchorList, CancellationToken cancellationToken)
+        {
+            int baseSize = start == null ? 0 : anchorList[(DateTimeOffset)start];
+            var last = start;
+            foreach (var item in anchorList)
```

Comment:
the archorList is added in BisectAnchor. Will it occur issus to use foreach here?

Reply:
Will not continue loop when jump into BisectAnchor function.